[studio-ui] sidebar updates

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/4351
https://github.com/craftercms/craftercms/issues/4358
https://github.com/craftercms/craftercms/issues/4353

aachavarria committed:
Changes on Widget.tsx to support sidebar.xml config;

aachavarria committed:
permittedRoles validation;

rart commented on ui/app/src/components/Navigation/PathNavigator/Widget.tsx:
```diff
@@ -85,12 +84,22 @@ const menuOptions = {
   }
 };
 
+interface OverrideClass {
```
StateStylingProps

rart commented on ui/app/src/components/Navigation/PathNavigator/Widget.tsx:
```diff
@@ -124,10 +133,11 @@ export interface WidgetState {
   const state = useSelection((state) => state.pathNavigator)[id];
   const itemsByPath = useSelection((state) => state.content.items).byPath;
   const site = useActiveSiteId();
+  const locale = 'en';
```
Locale should still be a prop. If not supplied, use default locale

rart commented on ui/app/src/components/Navigation/PathNavigator/Widget.tsx:
```diff
@@ -389,9 +401,29 @@ export function WidgetUI(props: any) {
+        style={{
+          ...icon.baseStyle,
+          ...(state?.collapsed ? icon.collapsedStyle : icon.expandedStyle)
+        }}
```
- Is state ever undefined here? Perhaps we need a wrapping loading state or resource?
- If state is ever undefined, these conditions would always use the expanded style, which I'm not sure that's right

aachavarria replied:
state is never undefined, this was my mistake. I will remove state validation.

rart commented on ui/app/src/modules/Preview/Tools/SiteExplorer.tsx:
```diff
@@ -84,33 +85,36 @@ const useExplorerStyles = makeStyles(() =>
-  'site-config': SiteConfigLink,
-  dashboard: DashboardLink
+  'craftercms.pathNavigator': Widget,
+  'craftercms.link': LinkWithIcon
```
craftercms.link => craftercms.linkWithIcon

rart commented on ui/app/src/utils/xml.ts:
```diff
@@ -151,7 +151,7 @@ export function toJS(xml: string | XMLDocument): any {
     xml = serialize(xml);
   }
-  return parser.parse(xml, {});
+  return parser.parse(xml, { attributeNamePrefix:"", ignoreAttributes: false});
```
- Format(!!!!!)
- Add options prop

rart reviewed (changes requested):
Fix codacy

aachavarria committed:
PR Fixes;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/sidebarUpdates

 Conflicts:
	ui/app/src/services/configuration.ts

aachavarria committed:
Merge and prettier;

aachavarria committed:
updating localeCode from xml;
renaming craftercms.link to craftercms.linkWithIcon;

aachavarria committed:
prettier;

aachavarria committed:
updating widget to use excludes;
updating pathNavigator.ts to use getIndividualPaths instead of getParentsFromPath;
fix on getParentsFromPath not returning last path;

rart commented on ui/app/src/utils/xml.ts:
```diff
@@ -145,13 +148,13 @@ export function wrapElementInAuxDocument(element: Element): XMLDocument {
     xml = serialize(xml);
   }
-  return parser.parse(xml, {});
+  return parser.parse(xml, options);
```
Default these to the ones you had.
```js
{
  blah: blah,
  blah: blah,
  ...options
}
```

rart commented on ui/app/src/utils/xml.ts:
```diff
@@ -145,13 +148,13 @@ export function wrapElementInAuxDocument(element: Element): XMLDocument {
-export function toJS(xml: XMLDocument): any;
-export function toJS(xml: string | XMLDocument): any {
+export function toJS(xml: string, options?: X2jOptionsOptional): any;
+export function toJS(xml: XMLDocument, options?: X2jOptionsOptional): any;
```
Overrides should go as
```js
export function toJS(xml: string): any;
export function toJS(xml: XMLDocument): any;
export function toJS(xml: string, options: X2jOptionsOptional): any;
export function toJS(xml: XMLDocument, options: X2jOptionsOptional): any;
export function toJS(xml: string | XMLDocument, options?: X2jOptionsOptional): any {
  // ..implementation
```

aachavarria committed:
common-api.js, wcm-assets-folder.js: fixing rootPath on viewDependencies;
ItemMenu.tsx: sending rootPath to historyDialog and dependenciesDialog;
PR Review;

aachavarria committed:
implementing HistoryDialog on DependenciesDialog.tsx;

aachavarria committed:
adding reverse() to getIndividualPaths;

rart commented on static-assets/components/cstudio-common/common-api.js:
```diff
@@ -897,7 +897,7 @@ var nodeOpen = false,
       },
 
-      viewDependencies: function (site, items, approveType, defaultSelection) {
+      viewDependencies: function (site, items, approveType, defaultSelection, rootPath) {
```
Revert and extract from item

rart commented on static-assets/components/cstudio-common/common-api.js:
```diff
@@ -906,6 +906,7 @@ var nodeOpen = false,
           payload: {
             open: true,
             item: CrafterCMSNext.util.content.parseLegacyItemToSandBoxItem(items[0]),
+            rootPath,
```
Extract from item

rart commented on static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-assets-folder.js:
```diff
@@ -1741,7 +1741,9 @@ var storage = CStudioAuthoring.Storage;
-                        false
+                        false,
+                        null,
+                        oCurrentTextNode.instance.config.params.path
```
Revert & extract from item

rart commented on ui/app/src/components/Navigation/PathNavigator/Widget.tsx:
```diff
@@ -85,12 +85,23 @@ const menuOptions = {
+  rootPath: string;
+  excludes: string[];
+  locale?: string;
+  icon: StateStylingProps | React.ElementType;
```
`icon: Partial< StateStylingProps>;`

rart commented on ui/app/src/components/Navigation/PathNavigator/Widget.tsx:
```diff
@@ -399,23 +420,43 @@ export function WidgetUI(props: any) {
+        icon={
+          icon.baseClass
+            ? `${icon.baseClass} ${state.collapsed ? icon.collapsedClass : icon.expandedClass}`
+            : icon
```
Fix logic

rart commented on ui/app/src/modules/Preview/Tools/SiteExplorer.tsx:
```diff
@@ -84,33 +85,36 @@ const useExplorerStyles = makeStyles(() =>
+}
+
+const LinkWithIcon = (props: LinkWithIconProps) => {
+  const { link, label, altText, icon } = props;
```
Alt text not applicable to links

rart commented on ui/app/src/modules/Preview/Tools/SiteExplorer.tsx:
```diff
@@ -84,33 +85,36 @@ const useExplorerStyles = makeStyles(() =>
-const LinkWithIcon = ({ label, icon, href }) => {
+interface LinkWithIconProps {
+  label: string;
+  altText: string;
```
x

aachavarria committed:
revert on viewDependencies prop rootPath;

aachavarria committed:
PR Review;

aachavarria committed:
getSidebarItems reading site-explorer.xml

aachavarria committed:
adding defaults to icon and container

aachavarria committed:
form-engine.js: changing order of cancel and save buttons;

aachavarria committed:
[4367][3.1.x] Jump to Region does not work for features with internal names with space in Website Editorial

https://github.com/craftercms/craftercms/issues/4367

aachavarria committed:
Revert "[4367][3.1.x] Jump to Region does not work for features with internal names with space in Website Editorial"

This reverts commit a26caf73

aachavarria committed:
GlobalNav.tsx: reading site and global items from uiConfig state;
SiteExplorer.tsx: reading site explorer from uiConfig State;
adding uiConfig to the GlobalState, Creating reducer/actions/epics;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/sidebarUpdates

aachavarria committed:
moving // eslint-disable-next-line

rart commented on ui/app/src/modules/Preview/PreviewConcierge.tsx:
```diff
@@ -156,6 +157,15 @@ export function PreviewConcierge(props: any) {
 
+  // region UI Config
+  useEffect(() => {
+    if(site){
```
Format

rart commented on ui/app/src/modules/Preview/PreviewConcierge.tsx:
```diff
@@ -156,6 +157,15 @@ export function PreviewConcierge(props: any) {
+  // region UI Config
+  useEffect(() => {
+    if(site){
+      dispatch(fetchSiteUiConfig({ site }))
```
Semi

rart commented on ui/app/src/modules/Preview/PreviewConcierge.tsx:
```diff
@@ -156,6 +157,15 @@ export function PreviewConcierge(props: any) {
+    if(site){
+      dispatch(fetchSiteUiConfig({ site }))
+    }
+  }, [dispatch, site])
```
semi

rart commented on ui/app/src/modules/Preview/Tools/SiteExplorer.tsx:
```diff
@@ -14,30 +14,23 @@
 import makeStyles from '@material-ui/core/styles/makeStyles';
 import ToolPanel from './ToolPanel';
-import { SidebarConfigItem } from '../../../services/configuration';
+import { siteExplorerItem } from '../../../services/configuration';
```
Interface name should be capitalised

rart commented on ui/app/src/services/configuration.ts:
```diff
@@ -28,17 +28,24 @@ import ContentType, { ContentTypeField } from '../models/ContentType';
+): Observable<string> {
+  return get(
+    `/studio/api/2/configuration/get_configuration?siteId=${site}&module=${module}&path=${configPath}`
+  ).pipe(map(({ response }) => response.content));
```
use pluck(response, content)

rart commented on ui/app/src/services/configuration.ts:
```diff
@@ -62,14 +69,14 @@ const LegacyPanelIdMap: any = {
   'component-panel': 'craftercms.ice.components',
   'medium-panel': 'craftercms.ice.simulator',
-  'targeting': 'craftercms.ice.audiences'
+  targeting: 'craftercms.ice.audiences'
```
let’s talk about removing all these mappings, being a brand new config, we don’t need to map legacy stuff

aachavarria committed:
PR changes;

aachavarria committed:
GlobalNav.tsx: Reverting changes;
ToolbarGlobalNav.tsx: Reverting changes;
GlobalState.ts: adding preview to uiConfig;
Renaming PreviewTool definitions;
ToolsPanel.tsx: Reading panels from store;
Deleting fetchPreviewToolsConfig and others;

aachavarria committed:
moving code;

aachavarria committed:
adding systemLinks to GlobalNav.tsx and SiteExplorer.tsx;
Parsing siteNavLinks;

aachavarria committed:
renaming channels to devices;

rart committed:
Elaborate comment

aachavarria committed:
removing toolbar from uiConfig.ts

aachavarria committed:
Merge remote-tracking branch 'aaron/feature/sidebarUpdates' into feature/sidebarUpdates

rart commented on ui/app/src/services/configuration.ts:
```diff
@@ -258,67 +197,77 @@ function deserializeActiveTargetingModelData<T extends Object>(data: T, contentT
+          }
+        };
       } else {
         return [];
```
Incorrect.

In fact, for now, if the file is missing, while we have a UM, return a mockup with the typical values.

rart commented on ui/app/src/modules/Preview/Tools/SiteExplorer.tsx:
```diff
@@ -84,33 +88,59 @@ const useExplorerStyles = makeStyles(() =>
+  return (
+    <LinkWithIcon
+      link="/studio/dashboard"
+      label={formatMessage(translations['dashboard'])}
```
use dot notation

rart commented on ui/app/src/modules/Preview/Tools/SiteExplorer.tsx:
```diff
@@ -84,33 +88,59 @@ const useExplorerStyles = makeStyles(() =>
+  return (
+    <LinkWithIcon
+      link="/studio/site-config"
+      label={formatMessage(translations['siteConfig'])}
```
use dot notation

aachavarria committed:
adding uiConfigMock.ts;

aachavarria committed:
using dot notation

rart committed:
Update uiConfigMock.ts

rart committed:
Update configuration.ts

sumerjabri commented:
@aachavarria please fix conflicts.

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/sidebarUpdates

 Conflicts:
	ui/app/src/components/Navigation/PathNavigator/PathNavigatorHeader.tsx
	ui/app/src/components/Navigation/PathNavigator/Widget.tsx
	ui/app/src/services/configuration.ts
	ui/app/src/services/content.ts
	ui/app/src/utils/xml.ts

aachavarria committed:
Merge remote-tracking branch 'aaron/feature/sidebarUpdates' into feature/sidebarUpdates
